### DISCUSSION TITLE
Beacon support

### DISCUSSION THREAD

**Discussion by bodeg** [dotnet-bluetooth-le/dotnet-bluetooth-le#789] [Q&A]
Can I use the plugin for iBeacon scanning?

All I need is to discover the beacon, and retrieve its Major and Minor numbers both on Android and iOS.

Thanks for any help

**Comment by fusionconsulting** [comment]
Did you get anywhere with this?

**Comment by axa88** [comment]
> Can I use the plugin for iBeacon scanning?

Yes

> All I need is to discover the beacon, and retrieve its Major and Minor numbers both on Android and iOS.

You will seek the `AdvertisementRecordType.ManufacturerSpecificData` withing the `AdvertisementRecords` of your scanned `IDevice`.
Then extract the version info from the 22nd to 25 byte of the `AdvertisementRecord.Data` property.

You might limit the the returned results to only Apple by applying a manufacturer Scan Filter or test upon it being returned from the same `Data` property in bytes 2 and 3.

> Thanks for any help

you're welcome
